src/settings/settings.controller.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  Controller,
+  Post,
+  Put,
+  Delete,
+  Param,
+  Body,
+  UseGuards,
+  ParseIntPipe,
+} from '@nestjs/common';
+import { SettingsService } from './settings.service';
+import { CreateSettingDto } from './dto/create-setting.dto';
+import { UpdateSettingDto } from './dto/update-setting.dto';
+import { Setting } from './models/setting.model';
+import { BasicAuthGuard } from 'src/auth/basic-auth.guard';
+
+@Controller('settings')
+@UseGuards(BasicAuthGuard)
+export class SettingsController {
+  constructor(private readonly settingsService: SettingsService) {}
+
+  @Post()
+  create(@Body() createSettingDto: CreateSettingDto): Promise<Setting> {
+    return this.settingsService.create(createSettingDto);
+  }
+
+  @Put(':id')
+  update(
+    @Param('id', ParseIntPipe) id: number,
+    @Body() updateSettingDto: UpdateSettingDto,
+  ): Promise<Setting> {
+    return this.settingsService.update(id, updateSettingDto);
+  }
+
+  @Delete(':id')
+  remove(@Param('id') id: number): Promise<void> {
```

Comment:
```suggestion
  remove(@Param('id', ParseIntPipe) id: number): Promise<void> {
```
Please check the code in all controllers to ensure the parsing is not skipped.